packages/devextreme/js/__internal/core/license/license_validation.ts

```diff
@@ -78,3 +82,53 @@ export function parseToken(encodedToken: string | undefined): Token {
     },
   };
 }
+
+export type LicenseVerifyResult = 'W0019' | 'W0020' | 'W0021' | null;
+let isLicenseVerified = false;
+
+const verifyLicense = function verifyLicense(licenseToken: string, ver: string = version): void {
+  if (isLicenseVerified) {
+    return;
+  }
+  isLicenseVerified = true;
+
+  let warning: LicenseVerifyResult = null;
+
+  if (licenseToken) {
+    const license = parseToken(licenseToken);
+
+    if (license.kind === CORRUPTED) {
+      warning = 'W0021';
+    } else {
+      const [major, minor] = ver.split('.').map(Number);
+
+      if (major * 10 + minor > license.payload.maxVersionAllowed) {
+        warning = 'W0020';
+      }
```

Comment:
If we will have ```ver = '123'``` we verify the license without any error

Because the ```123 * 10 + NaN = NaN```
And ```NaN > 10000000``` is false

Therefore, we should add an additional condition here.
Maybe we need the special error code for it :)

Reply:
I reused code W0021 because otherwise I would have to agree with Ray about 1 more error text.